petastorm/unischema.py

```diff
@@ -165,13 +187,41 @@ def as_spark_schema(self):
         >>> spark.createDataFrame(dataset_rows,
         >>>                       SomeSchema.as_spark_schema())
         """
-        schema_entries = [
+        schema_entries = []
+        in_bundles = set()
+        if self._column_bundles:
+            # If we are using column bundles, we add a schema entry for each bundle and keep track
+            # of which columns are in bundles to add them individually
+            for bundle_name, bundle_columns in self._column_bundles.items():
+                schema_entries.append(StructField(bundle_name, BinaryType(), False))
+                in_bundles.update(bundle_columns)
+
+        # We add schema entries for each field that is not already in a bundle
+        schema_entries.extend([
             StructField(
                 f.name,
                 f.codec.spark_dtype(),
-                f.nullable) for f in self._fields.values()]
+                f.nullable) for f in self._fields.values() if f.name not in in_bundles])
         return StructType(schema_entries)
 
+    def get_column_names(self):
+        """Gets a set of column names taking into account column bundles"""
+        # Build a dictionary of field to bundle name it occurs in
+        # Since a field can appear in multiple bundles, a random one will be chosen
+        field_to_bundle_name = {}
```

Comment:
Maybe rewriting it using set operations could be slightly more concise/clear (or maybe I am just trying to be a smartass)?
```
column_bundles  = {'b0': ['a', 'b', 'c'], 'b1': ['d']}
fields = {'a', 'b', 'c', 'd', 'e'}
bundled_fields = set(itertools.chain(*column_bundles.values()))
all_columns = set(column_bundles.keys()) | (fields - bundled_fields)
```

Reply:
Yes, that does seem like a better way to write this